src/amber/support/file_encryptor.cr

```diff
@@ -1,29 +1,26 @@
 require "./message_encryptor"
 
 module Amber::Support
-  module FileEncryptor
-    def self.encryption_key
-      ENV["AMBER_ENCRYPTION_KEY"]? || begin
-        if File.exists?(".encryption_key")
-          File.open(".encryption_key").gets_to_end.to_s
-        else
-          raise Exceptions::EncryptionKeyMissing.new
-        end
-      end
-    end
+  SECRET_KEY  = "AMBER_ENCRYPTION_KEY"
+  SECRET_FILE = "./.encryption_key"
 
-    def self.read(filename : String, encryption_key = self.encryption_key)
+  module FileEncryptor
+    def self.read(filename : String, encryption_key = encryption_key)
       encryptor = Amber::Support::MessageEncryptor.new(encryption_key)
       encryptor.verify_and_decrypt(File.open(filename).gets_to_end.to_slice)
     end
 
-    def self.write(filename : String, body : (String | Bytes), encryption_key = self.encryption_key)
-      encryptor = Amber::Support::MessageEncryptor.new(encryption_key)
+    def self.write(filename : String, body : (String | Bytes), encryption_key = encryption_key)
+      encryptor = MessageEncryptor.new(encryption_key)
       File.write(filename, encryptor.encrypt_and_sign(body))
     end
 
-    def self.read_as_string(filename, encryption_key = self.encryption_key)
+    def self.read_as_string(filename, encryption_key = encryption_key)
       String.new(read(filename, encryption_key))
     end
+
+    private def self.encryption_key
+      ENV[SECRET_KEY]? || File.open(SECRET_FILE).gets_to_end.to_s
```

Comment:
Why does this no longer raise an exception? If it tries reading a file that doesn't exist it will anyway. Since we're loading settings with cli now instead of for specific commands this will throw an exception unexpectedly.

Lets pair on this.

Reply:
One of the reasons that I went with a class was because of this and the encryption_key default value. With the class it guarantee to have a key before using the FileEncryptor and there is no assumptions made about a default value for the key

Reply:
True but that's just taking the problem and moving it further out from the center. It still may or may not exist and needs to be dealt with regardless of where we put it.

Moving it further from it's use case just adds complexity to the problem.